integrations/snyk/main.py

```diff
@@ -71,14 +74,20 @@ async def on_projects_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
 
     semaphore = asyncio.Semaphore(CONCURRENT_REQUESTS)
     async for projects in snyk_client.get_paginated_projects():
-        logger.debug(
-            f"Received batch with {len(projects)} projects, getting their issues"
-        )
-        tasks = [process_project_issues(semaphore, project) for project in projects]
-        issues = await asyncio.gather(*tasks)
-        yield [
-            {**project, "__issues": issues} for project, issues in zip(projects, issues)
-        ]
+        logger.debug(f"Received batch with {len(projects)} projects")
+
+        if cast(
+            ProjectResourceConfig, event.resource_config
+        ).selector.attach_issues_to_project:
+            logger.debug("Getting issues for projects in batch")
+            tasks = [process_project_issues(semaphore, project) for project in projects]
+            issues = await asyncio.gather(*tasks)
+            yield [
+                {**project, "__issues": issues}
+                for project, issues in zip(projects, issues)
+            ]
```

Comment:
lets add a warning when user uses that flag, to indicate that it will be deprecated, as I think this is not too efficient, and also we want users to map the data differently